src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java

```diff
@@ -12,11 +18,39 @@ public FigureDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public T save(T figure) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(figure);
+            transaction.commit();
+            return figure;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert figure entity "
```

Comment:
```suggestion
            throw new RuntimeException("Can't insert figure entity " + figure, e);
```
we can inline if checkstyle allows